src/ClientFactory.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace oliverlorenz\reactphpmqtt;
+
+use oliverlorenz\reactphpmqtt\protocol\Version;
+use React\Dns\Resolver\Factory as DnsResolverFactory;
+use React\EventLoop\Factory as EventLoopFactory;
+
+class ClientFactory
+{
+    public static function createClient(Version $version, $resolverIp)
```

Comment:
can you give $resolverIp a default?

BTW: This resolver thing is something I'm not happy with, because its a unusual thing you have to set your DNS manually. Maybe you have a idea to change that? But, if we do that lets do this in another PR